### ISSUE TITLE
Add custom properties to be recognized

### ISSUE THREAD

**Issue by mradke** [stylelint/stylelint#1903]
This issue is probably a feature request. I want to be able to specify custom properties without deactivating "property-no-unknown" completely.

Given the following post-css (I am using https://github.com/peterramsing/lost) declaration:
```css
.foo {
  lost-utility: clearfix;
}
```

and this stylelint config
```json
{
  "rules": {
    "property-no-unknown": true,
    "custom-property-pattern": "lost-.+",
  }
}
```

Running Version `7.2.0` through Webpack and Atom plugin should not result in a warning or error.

However I am issued with an error `Unexpected unknown property "lost-utility" (property no unknown)`.

Probably my understanding for the `custom-property-pattern` function is wrong. How do others handle this?

**Comment by evilebottnawi** [comment]
@mradke `property-no-unknown` and `custom-property-pattern` is independent rules, you should add `ignoreProperties` option for excluded `lost-utility` http://stylelint.io/user-guide/rules/property-no-unknown/#ignoreproperties-regex-string

**Comment by mradke** [comment]
Ah thanks for the heads up! I was completely missing the secondary options of the rule - sorry.

**Comment by sergeysova** [comment]
What about nested props?

```scss

:root {
  --color: {
    red: #F15757;
    green: #2CDEC0;
    blue: #1b96f5;
    gray: #D8D8D8;
    grayText: #999999;
    lightGray: #B3B3B3;
  }
}

```
